### DISCUSSION TITLE
∞ frame limit

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#26637] [Ideas] [answered]
A small issue in osulazer. Then I explained whether the 'unlimited' option should be added.Finally, I attempted to prove that 'latency certification' is meaningless.
--

Just these three.

**The issue with the 'latency certifier' interface in osulazer**
--

In the 'latency certifier' screen, adjusting the frame limit (first VSync, then Basically Unlimited) will result in an unlimited frame limit for Draw and Update. After returning to the game, Draw still displays 1000/∞ fps, but actually it may still be limited to 1000

If in 'single thread', the input will display as unlimited upon returning to the game, but there is no confirmation as I am unable to reach 900fps.

So, we only obtained an unrestricted frame rate in the 'latency verifier' and an error display in the game. In the 'latency verifier' screen with unlimited frame rates, you will feel a slight decrease in latency compared to in the game (which appears to be due to 'Update').

If it is already displayed as 1000/∞ fps, then the fps item is always red.

**Why do I think 'unlimited' should be added**
--

In OSU stable, you can choose 'unlimited', which has been around for a long time.

Although 'unlimited' may lead to even worse situations But it's not always like this, so I think it should exist, warning players when they choose 'unlimited'.

Players will not be unaware of these differences in latency, especially when comparing between 10000fps and 1000fps (10000fps after lifting restrictions in osulazer). I feel that there are indeed differences, and they may be more pronounced during gameplay.

It may indeed have negative impacts, so a warning should be added when selecting 'unlimited'. This is a compromise idea.

'latency certifier' is meaningless.
--

Can 'latency verifier' really be tested? Even in 'level 1', it looks like the delay on both sides is the same, only the draw rate is different. This makes it difficult for players to feel the difference after exceeding the player's display resolution. Does this mean that I only need to set it to VSync?
My proof is that when you adjust the frame limit in the 'latency certifier' and compare the delay on the side of the smaller fps, you will find that the difference is very large, even at the same 'level'.

If you want to reproduce it, it is recommended to go to level 2 because the draw fps is a bit higher to prevent it from affecting your judgment. The difference between VSync and Basically unlimited can be clearly felt, and 8x refresh rate may be difficult to see the difference because the draw fps is too low (you can reach the high 'level' first before trying again, but because the delay has not changed, it may be difficult to reach the 'level' beyond the display frame rate. My method is to see the picture tearing on one side).

**Comment by GopoBoi** [comment]
Warning players about the issues with Unlimited frame limiter doesn't work, as there is a warning in stable yet people still use it even when it causes stutters.
Besides there really is no point in having over 1000 fps. even in your extreme example of 10000 fps vs 1000 fps that is a difference of *at worst* 0.9 ms, that is incomprehensible to humans. And also there are many things adding to latency where the average player is probably playing with around 25 - 40 ms of latency, so in comparison that 0.9 ms is absurdly insignificant.

**Reply by nana123126** [reply]
Okay, since you think so. Perhaps I am too sensitive, or is it only happening on my device? There may also be other reasons. Anyway, I think. I see a higher 'Update' for a better experience.

**Comment by sineplusx** [comment]
I agree with some of your points. The latency certifier is kinda useless considering how broken and cheesable it is. The methodology in the circle mode of the certifier does not involve latency at all, but it is designed to help players choose a frame limiter that does not affect their perception of gameplay.

However, the points outlined in the wiki article still stand, and at this point I think you are placebo-ing yourself. There are better ways to improve perceptual latency than letting your graphics card whine in agony.

**Reply by nana123126** [reply]
Okay, I think it's a 'slight' decrease in latency. But if that's the case, I think osulazer should add a reliable vertical synchronization, because I've found that everything except for 'VSync' can cause screen tearing, and the higher the frame rate, the less obvious the tearing. Perhaps I feel a decrease in latency because of this?

I won't use 'VSync' frame limiting to avoid screen tearing because it has significant latency.

**Comment by peppy** [comment, marked as answer]
Sorry but we've been through this many times.

See https://github.com/ppy/osu/wiki/Latency-and-unlimited-frame-rates.
